backend/src/lib/emailVerificationPlugin.js

```diff
@@ -0,0 +1,39 @@
+const nanoid = require('nanoid').customAlphabet('0123456789abcdefghijklmnopqrstuvwxyz', 40)
+const { sendEmailVerification } = require('../email')
+
+// Must called after passport-local-mongoose applied to schema, because this assumes there is an authenticate instance method.
+module.exports = function (schema) {
+  schema.add({
+    isVerified: {
+      type: Boolean,
+      default: false,
+    },
+    emailVerificationToken: {
```

Comment:
Token should be removed after verification

Create the token when sending the email

Reply:
Token should have a timeout

Reply:
Token should be hold in a different collection

Reply:
Didn't implement timeout for now

Reply:
implemeted :)